### DISCUSSION TITLE
[New] Proposition d'ajout de pratique : « Favoriser le libre (open source) et contribuer aux communs »

### DISCUSSION THREAD

**Discussion by florinesueur** [cnumr/best-practices#31] [Bonnes pratiques] [answered]
A mon sens, le libre permet d'allonger la durée de vie des appareils, tout en réduisant le temps de développement pour une utilisation du plus grand nombre et mondialement.

**Comment by ACTLEM** [comment]
Nous sommes entièrement d'accord là dessus. La question que je me pose est "est-ce que cela doit figurer dans un référentiel d'écoconception web?". Pour moi cette pratique a plus sa place dans un référentiel GreenIT/Numérique Responsable, à un niveau plus large.
Je proposerai plus une tournure du style: "Eviter d'inclure des framework, outils, solutions propriétaires au sein de votre architecture (sous entendu web)" ou encore "Utiliser que des dépendances open sources dans l'implémentation de votre service"...
Y a sûrement mieux que ces deux propositions, mais c'est un peu l'idée que je vois.

**Reply by arnolem** [reply]
Je me suis fait la même remarque.
Un exemple : Si les fabricants de materiel mettaient en OpenSource leur DRIVERS dès qu'ils en arrêtent le support, la communauté pourrait continuer de maintenant un matériel obsolète sur les derniers OS.

Mais dans le Web, j'ai déjà vu des exemples qui ne me semblent pas pertinent pour de l'éco-conception.
Par exemple : 
 - Remplacer Google Map par OpenStreetMap (en quoi c'est de l'éco-conception ? )
 - Remplacer Googla Analytics par Matomo (en quoi c'est de l'éco-conception ? )
 
 Ce sont pourtant deux exemples que j'ai déjà vu cité dans un conférence sur l'éco-conception sans comprendre le "Pourquoi".

**Reply by ACTLEM** [reply]
Google Maps abandonne des fonctionnalités, des versions (ex V2 Javascript depuis Mai 2021). Cela peut provoquer une obsolescence d'un service qui se repose sur les fonctionnalités abandonnées par exemple et donc un service ou une partie du service qui devient inutilisable.
C'est peut-être pour cela que cela a été mentionné dans une conférence.

**Reply by arnolem** [reply]
Si on prend cet axe, quel est le problème de l'obsolescence d'un service ?
Si le service n'est plus fonctionnel, il ne sera plus utilisé et sera peut-être même désactivé ?
Si un service disparait, c'est bien eco-responsable ?

**Reply by ACTLEM** [reply]
Un service inutilisable même partiellement génère une consommation de ressources inutiles car il ne peut plus rendre le service attendu.
Le fait de décommissionner un service doit être un choix métier (premières bonnes pratiques) et non subi par un choix technique.

**Reply by DocRoms** [reply]
> Mais dans le Web, j'ai déjà vu des exemples qui ne me semblent pas pertinent pour de l'éco-conception. Par exemple :
> 
> * Remplacer Google Map par OpenStreetMap (en quoi c'est de l'éco-conception ? )
> * Remplacer Googla Analytics par Matomo (en quoi c'est de l'éco-conception ? )
> 
> Ce sont pourtant deux exemples que j'ai déjà vu cité dans un conférence sur l'éco-conception sans comprendre le "Pourquoi".

Je penses que nous sommes également sur un problèmes de wording, il y a beaucoup de personnes qui confondent "`éco-conception`" et "`numérique responsable`". 
    
 - D'un point de vue éco-conception, les services de googles peuvent être utiles: ils peuvent être bien conçus, paramétrables pour éviter de surconsommer des ressources inutiles.... Quand ils ne téléchargent pas la terre entière de JS pour tracker la moindre chose, en ne leur ayant rien demandé 😅 .
 - D'un point de vue numérique responsable, par contre, filer toutes ces données à Google avec toutes les affaires qui leur court après, c'est loin d'être une bonne idée, à moins de se foutre totalement de ce que nos Thirds Parts font avec les données de nos utilisatrices et utilisateurs.

Pour ce qui est de l'obsolescence je rejoins les propos de @ACTLEM quand à la résilience du logiciel suite à une interruption de services  de la tierce partie utilisée : mieux vaut utiliser des services éprouvés qui ne cesserons pas de fonctionner du jour au lendemain, que d'utiliser un service qui ne sera peut être plus maintenu dans les mois/années qui viennent. 

Par contre vous auriez des sources tous les deux ? (#curieux)
 - @arnolem sur la conf ou ils parlent de remplacer gA et gMaps par des outils alternatifs
 - @ACTLEM sur le soucis de la v2 JS de gMap .

**Reply by ACTLEM** [reply]
https://developers.google.com/maps/deprecations?hl=fr#maps_javascript_api_v2_may_19_2010_-_may_26_2021

**Reply by tbroyer** [reply]
Le fait que Google supprime la v2, ça a quel impact concrètement ?

* le service qui en dépendait doit être mis à jour (la v3 est stable depuis plus de 10 ans, est-ce que ça aurait pas pu/du être fait avant ?), ça va demander des resources de dév, et avoir un impact sur le caching des resources Web. Quelqu'un sait quel impact la v3 a sur l'usage des resources client et réseau et si finalement cette migration vers la v3 ne serait pas une bonne chose pour les utilisateurs ?
* le service qui en dépend ne fonctionne plus, et s'il n'est pas mis à jour (et si personne ne s'en est préoccupé durant les 10 dernières années), on pourrait considérer que le service aurait dû être décommissionné plus tôt finalement !

Mais finalement on parle là principalement d'une dépendance à un service externe, qui n'a pas grand chose à voir avec le fait qu'il soit Open Source ou non.

Si on se recentre sur la question du Libre, je rejoins complètement @DocRoms que l'impact sur l'écoconception n'est pas du tout évident (et pourrait même être négatif) et que ça s'inscrit plutôt dans une démarche de numérique responsable au sens large (et encore, ça reste je pense discutable) où on acceptera par exemple de faire des compromis sur l'impact éco du service parce qu'il y a d'autres arguments en faveur de cette solution technique qui nous font la préférer à une autre.

**Reply by arnolem** [reply]
>     * @arnolem sur la conf ou ils parlent de remplacer gA et gMaps par des outils alternatifs

Matomo : https://clips.twitch.tv/DiligentIronicMooseLitFam-MOZSA1F6jzLz83yh
OpenStreetMap : https://clips.twitch.tv/MoldyGrossRatMikeHogu-7c5G7eAzriylZULf

La conférence est sur twitch pour quelques jours et sera sur youtube par la suite : https://www.twitch.tv/videos/1174772478
"Introduction au design de services numériques éco-responsables - Aurélie Baton/Anne Faubry"
Attention, je ne résume pas la conférence à ces deux extraits, c'est simplement pour illustrer ce type de "bonnes pratiques open source" sans argument eco.

**Reply by ACTLEM** [reply]
D'accord avec @tbroyer et @DocRoms sur le côté numérique responsable au sens large, cela rejoint ma première remarque :-)
Sur cette proposition, je pense que l'on manque d'éléments probants pour faire rentrer dans l'écoconception web même si pour d'autres catégories de services (ex: basée sur des objets connectés comme le dit @arnolem), c'est déjà un peu plus facile.

**Reply by florinesueur** [reply]
D'après une étude de la CNIL :
"Sur le plan environnemental :
   – La transparence du code donne des leviers d’action pour l’écoconception, au-delà d’une simple mesure de consommation
  – Mutualiser et recycler du code permet une plus grande sobriété énergétique par rapport à des entreprises propriétaires dont chacune (re)développe son code."
[Source](https://cnll.fr/media/etude-cnll-2021.pdf)

Quand j'ai écrit cette proposition, je pensais à :
 - Sensibiliser, évangéliser (via des issues ou contributions) les dev à l'écoconception en intégrant un projet open source ou en utilisant un (exemple : des propositions de wordpress plus écoconçu)
 - Il y a peu, j'ai travaillé sur une PWA avec une fonctionnalité de transcription Speech-to-Text (malheureusement indispensable). Au lieu d'utiliser des API de google, aws, azure, ... nous avons préféré nous tourner vers Modzilla et contribuez à Common Voice qui plus ici des data qui tombent dans le domaine public. (open data)
 - et comme l'a indiqué @ACTLEM, éviter l'obsolescence des terminaux avec une nouvelle version d'une solution propriétaire non modifiable et potentiellement plus gourmande.

Je comprends tout à fait que cette proposition relève plus du numérique responsable et elle manque de données.

**Comment by ACTLEM** [comment, marked as answer]
Note Equipe Projet: L'open source en soit ne permet pas de garantir une réduction des impacts même si nous sommes d'accord que pousser l'open source est une pratique que nous soutenons.

**Reply by Christophe-Cl** [reply]
Totalement d'accord ! À fond pour l'opensource et le libre, mais il existe trop de contre-exemples d'outils générant plus d'impact environnemental qu'une solution propriétaire.
